br/pkg/lightning/backend/external/file.go

```diff
@@ -46,51 +49,26 @@ func NewKeyValueStore(
 	return kvStore, nil
 }
 
-// AddKeyValue saves a key-value pair to the KeyValueStore. If the accumulated
+// AddData saves encoded key-value pairs to the KeyValueStore.
+// data layout: keyLen + key + valueLen + value. If the accumulated
 // size or key count exceeds the given distance, a new range property will be
 // appended to the rangePropertiesCollector with current status.
 // `key` must be in strictly ascending order for invocations of a KeyValueStore.
-func (s *KeyValueStore) AddKeyValue(key, value []byte) error {
-	var (
-		b     [8]byte
-		kvLen = 0
-	)
-
-	// data layout: keyLen + key + valueLen + value
-	n, err := s.dataWriter.Write(
-		s.ctx,
-		binary.BigEndian.AppendUint64(b[:0], uint64(len(key))),
-	)
-	if err != nil {
-		return err
-	}
-	kvLen += n
-	n, err = s.dataWriter.Write(s.ctx, key)
-	if err != nil {
-		return err
-	}
-	kvLen += n
-	n, err = s.dataWriter.Write(
-		s.ctx,
-		binary.BigEndian.AppendUint64(b[:0], uint64(len(value))),
-	)
-	if err != nil {
-		return err
-	}
-	kvLen += n
-	n, err = s.dataWriter.Write(s.ctx, value)
+func (s *KeyValueStore) AddData(val []byte) error {
```

Comment:
Just noticed no other package is using this function so we can make it private. Another reason is now the function need caller to build data in specific layout, which is not friendly as a public function.

Reply:
https://github.com/pingcap/tidb/pull/47793/commits/c8e33104edb36bdca262e713451c77a20dca0953